File: core/src/test/java/de/qabel/core/repository/EntityManagerTest.java
package de.qabel.core.repository;

import de.qabel.core.TestServer;
import de.qabel.core.config.Identity;
import de.qabel.core.config.factory.DropUrlGenerator;
import de.qabel.core.config.factory.IdentityBuilder;
import org.junit.Test;

import java.net.URISyntaxException;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class EntityManagerTest {
    private EntityManager em = new EntityManager();

    @Test
    public void containsNothingOnStart() {
        assertFalse(em.contains(Identity.class, 1));
    }

    @Test
    public void containsContainedEntity() throws URISyntaxException {
        Identity identity = new IdentityBuilder(new DropUrlGenerator(TestServer.DROP)).withAlias("test").build();
        identity.setId(1);
        em.put(Identity.class, identity);
        assertTrue(em.contains(Identity.class, 1));
    }

    @Test
    public void containsNothingAfterClear() throws Exception {
        Identity identity = new IdentityBuilder(new DropUrlGenerator(TestServer.DROP)).withAlias("test").build();
        identity.setId(1);
        em.put(Identity.class, identity);
        em.clear();
        assertFalse(em.contains(Identity.class, 1));
    }
}

File: core/src/test/java/de/qabel/core/repository/EntityManagerTest.kt
package de.qabel.core.repository

import de.qabel.core.config.Identity
import de.qabel.core.config.factory.DropUrlGenerator
import de.qabel.core.config.factory.IdentityBuilder
import de.qabel.core.repository.EntityManager
import org.junit.Test

import java.net.URISyntaxException

import org.junit.Assert.*

class EntityManagerTest {
    private val em = EntityManager()

    @Test
    fun containsNothingOnStart() {
        assertFalse(em.contains(Identity::class.java, 1))
    }

    @Test
    @Throws(URISyntaxException::class)
    fun containsContainedEntity() {
        val identity = IdentityBuilder(DropUrlGenerator("http://localhost")).withAlias("test").build()
        identity.id = 1
        em.put(Identity::class.java, identity)
        assertTrue(em.contains(Identity::class.java, 1))
    }

    @Test
    @Throws(Exception::class)
    fun containsNothingAfterClear() {
        val identity = IdentityBuilder(DropUrlGenerator("http://localhost")).withAlias("test").build()
        identity.id = 1
        em.put(Identity::class.java, identity)
        em.clear()
        assertFalse(em.contains(Identity::class.java, 1))
    }
}
